Repository/AuthenticationService.cs

```diff
@@ -1,57 +1,49 @@
+using GameStore.Dtos;
 using GameStore.Models;
 using GameStore.Shared.Helpers;
 using GameStore.Shared.Services;
-using Blazored.LocalStorage;
+using GameStore.Shared.States;
 using System.IdentityModel.Tokens.Jwt;
 using System.Security.Claims;
 
 namespace GameStore.Repository;
 
-public class AuthenticationService(IHttpService httpService, ILocalStorageService localStorage) : IAuthenticationService
+public class AuthenticationService(IHttpService httpService, LocalStorage localStorage) : IAuthenticationService
 {
-    public async Task<ResponseWrapper<Client>> Login(UserAccess userAccess)
+    public async Task<ResponseWrapper<LoginResponse>> Login(UserAccess userAccess)
     {
         try
         {
             var response = await httpService.Post<LoginResponse>("/api/users/authenticate", userAccess);
-            if (response.Token != null)
+            if (!string.IsNullOrEmpty(response.Token))
             {
-                await localStorage.SetItemAsync("authToken", response.Token);
-                var client = DecodeToken(response.Token);
-                return ResponseWrapper<Client>.Success(client);
-
+                await localStorage.SetToken(response.Token, "auth-token");
+                return ResponseWrapper<LoginResponse>.Success(response);
             }
-            return ResponseWrapper<Client>.Fail("Invalid response from server");
+            return ResponseWrapper<LoginResponse>.Fail("Invalid response from server");
         }
         catch (Exception e)
         {
             Console.WriteLine($"Access denied. Exception: {e.Message}");
-            return ResponseWrapper<Client>.Fail("Access denied");
+            return ResponseWrapper<LoginResponse>.Fail("Access denied");
         }
     }
 
-    public async Task Logout()
-    {
-        await localStorage.RemoveItemAsync("authToken");
-    }
-
-    private static Client DecodeToken(string token)
+    public static Client DecodeToken(string token)
```

Comment:
Fungsi ini mungkin lebih baik ada di AuthState class

Reply:
yes sempat kepikiran begitu juga, karena yang pakainya pun di AuthState dan kalau dianalisa kembali di FE ini benar kalau lebih dekat responsibilitynya dengan AuthState ketimbang AuthenticationService

Action Plan: move to AuthState